### PR TITLE
executor: remove Next() function for SelectionExec

### PR DESCRIPTION
**Pull request by qxhy123**

### CODE REVIEW

File: table/tables/tables_test.go [hunk 1]
```diff
@@ -184,9 +184,11 @@ func (ts *testSuite) TestTypes(c *C) {
 	rs, err = ts.se.Execute(ctx, "select c1 + 1 from test.t where c1 = 1")
```
Comment (zz-jason) [inline table/tables/tables_test.go]: ditto

File: executor/executor_test.go [hunk 2]
```diff
@@ -2263,9 +2263,12 @@ func (s *testSuite) TestBit(c *C) {
 	c.Assert(err, NotNil)
 	r, err := tk.Exec("select * from t where c1 = 2")
 	c.Assert(err, IsNil)
-	row, err := r.Next(context.Background())
+	//row, err := r.Next(context.Background())
```
Comment (XuHuaiyu) [inline executor/executor_test.go]: remove this line

File: executor/executor_test.go [hunk 3]
```diff
@@ -2263,9 +2263,12 @@ func (s *testSuite) TestBit(c *C) {
...
+	chk := r.NewChunk()
+	err = r.NextChunk(context.Background(), chk)
 	c.Assert(err, IsNil)
-	c.Assert(types.BinaryLiteral(row.GetBytes(0)), DeepEquals, types.NewBinaryLiteralFromUint(2, -1))
+	//c.Assert(types.BinaryLiteral(row.GetBytes(0)), DeepEquals, types.NewBinaryLiteralFromUint(2, -1))
```
Comment (XuHuaiyu) [inline executor/executor_test.go]: remove this line

File: table/tables/tables_test.go [hunk 4]
```diff
@@ -184,9 +184,11 @@ func (ts *testSuite) TestTypes(c *C) {
...
 	c.Assert(err, IsNil)
 	row, err = rs[0].Next(ctx)
+	chk := rs[0].NewChunk()
+	err = rs[0].NextChunk(ctx, chk)
 	c.Assert(err, IsNil)
-	c.Assert(row, NotNil)
-	c.Assert(row.GetFloat64(0), DeepEquals, float64(2))
+	c.Assert(chk.GetRow(0), NotNil)
```
Comment (XuHuaiyu) [inline table/tables/tables_test.go]: c.Assert(chk.NumRows() == 0, IsFalse)

File: table/tables/tables_test.go (hunk 1 above)
Comment (qxhy123) [reply table/tables/tables_test.go]: Do I need remove this line?

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason

File: table/tables/tables_test.go (hunk 1 above)
Comment (zz-jason) [reply table/tables/tables_test.go]: no, I think you should rewrite the test in this way:
```go
rs = tk.MustQuery("select c1 + 1 from test.t where c1 = 1")
rs.Check()
```

**Comment by zhexuany** [comment]
@qxhy123 Please resolve conflicts and address comment.

**Pull request merged**